### DISCUSSION TITLE
Open-source inquires

### DISCUSSION THREAD

**Discussion by abuibaid** [casadi/casadi#3286] [General]
Hi everyone,  
we have developed  software which is dependent on CasADI and IPOPT. We need to publish the code here in github with licesne. I know CasADI's license has a copyleft, as well as the ipopt which has weak copyleft.  Actually there is no modification occured on whether CasADI or IPOPT, while the aim is to redistribute the software which is dependent on these tools. The questions are : 1) how much freedom do I have without violating the copyleft? 2) are there any compatible licenses with CasADI or I can choose whatever I want?

I hope you guys explain this to me with simple terms :D :D :D 

Regards,

**Comment by jgillis** [comment]
(I'm not a lawyer)

You can distribute your casadi-dependent code using basically any license (also commercial) if you use casadi as a library. You can also distribute casadi itself or its binaries, provided that you retain license and copyright notice.

Some pointers: https://choosealicense.com/

What would it be like to use casadi not as a library: making modifications to it, linking statically with it, copy-pasting parts of the code into your project.
